Project2/Part_B/Deck.cpp

```diff
@@ -59,7 +59,28 @@ void Deck::shuffle(void)
   }
 }
 
-std::ostream& operator<<(std::ostream& os, Deck rhs)
+Card* Deck::deal(void)
+{
+  Card* dealt = this->header;
+  this->header = dealt->getNext();
+  dealt->setNext(nullptr);
+  return dealt;
+}
+
+void Deck::replace(Card* card)
+// this is a bit ugly but worked best for now while the copy constructor and assignment operator
+// are both in a different branch
+{
+  Card* head = this->header;
```

Comment:
also head is a terrible name for this because it is not the head and is never trying to be the head